### PR TITLE
build: bump up versions and dependencies

### PR DESCRIPTION
**Pull request by priteshbandi**
bump up versions and dependencies for rc6

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,16 +3,16 @@ module github.com/notaryproject/notation
...
 
 require (
-	github.com/notaryproject/notation-core-go v1.0.0-rc.3
-	github.com/notaryproject/notation-go v1.0.0-rc.5
+	github.com/notaryproject/notation-core-go v1.0.0-rc.4
+	github.com/notaryproject/notation-go v1.0.0-rc.6
 	github.com/opencontainers/go-digest v1.0.0
-	github.com/opencontainers/image-spec v1.1.0-rc.3
+	github.com/opencontainers/image-spec v1.1.0-rc3
 	github.com/oras-project/oras-credentials-go v0.1.1-0.20230517070118-6b1be42cc39f
```
Comment (yizha1) [inline go.mod:10]: There is new version of oras-credentials-go  `v0.1.1`, see https://github.com/oras-project/oras-credentials-go/releases/tag/v0.1.1

**Review by yizha1 — COMMENTED** [review]
Thanks @priteshbandi

**Review by byronchien — APPROVED** [review]

**Review by JeyJeyGao — APPROVED** [review]
LGTM

**Review by rgnote — APPROVED** [review]
LGTM

**Review by Two-Hearts — APPROVED** [review]
LGTM

File: go.mod [hunk 2]
```diff
@@ -3,16 +3,16 @@ module github.com/notaryproject/notation
...
 
 require (
-	github.com/notaryproject/notation-core-go v1.0.0-rc.3
-	github.com/notaryproject/notation-go v1.0.0-rc.5
+	github.com/notaryproject/notation-core-go v1.0.0-rc.4
+	github.com/notaryproject/notation-go v1.0.0-rc.6
 	github.com/opencontainers/go-digest v1.0.0
-	github.com/opencontainers/image-spec v1.1.0-rc.3
-	github.com/oras-project/oras-credentials-go v0.1.1-0.20230517070118-6b1be42cc39f
+	github.com/opencontainers/image-spec v1.1.0-rc3
```
Comment (Two-Hearts) [inline go.mod:9]: Sorry, just want to double check on this line of change. Is this correct? @priteshbandi @yizha1

Comment (JeyJeyGao) [reply go.mod:9]: Both the `v1.1.0-rc3` and the `v1.1.0-rc.3` tag to the same commit. According to the [release history](https://github.com/opencontainers/image-spec/releases) the image-spec usually use the style like `rc1`, `rc2`.

Comment (yizha1) [reply go.mod:9]: Thanks @JeyJeyGao It seems both versions work.

**Pull request merged**